### PR TITLE
Add Origin in DeletedDatabaseSpec

### LINKED ISSUE #29: Store original db spec in DDB's spec
```
spec:
  origin:
    metadata:
     ff
    spec:
      postgres:
        postgres_spec
      elastic:
```

### PR DESCRIPTION
**Pull request by aerokite**
Fixes #29

### CODE REVIEW

File: api/deleted_database_types.go [hunk 1]
```diff
@@ -28,6 +28,23 @@ type DeletedDatabaseSpec struct {
 	// If true, invoke recover operation
 	// +optional
 	Recover bool `json:"recover,omitempty"`
+	// Origin Spec to store original database information
+	Origin OriginSpec `json:"origin,omitempty"`
+}
+
+type OriginSpec struct {
```
Comment (TamalSaha) [inline api/deleted_database_types.go]: `OriginSpec` -> `Origin`

File: api/deleted_database_types.go [hunk 2]
```diff
@@ -28,6 +28,23 @@ type DeletedDatabaseSpec struct {
...
+	api.ObjectMeta `json:"metadata,omitempty"`
+	// OriginDatabase Spec to store original database Spec
+	OriginDatabase OriginDatabaseSpec `json:"originDatabase,omitempty"`
+}
+
+type OriginDatabaseSpec struct {
```
Comment (TamalSaha) [inline api/deleted_database_types.go]: `OriginDatabaseSpec` -> `OriginSpec`

File: api/deleted_database_types.go [hunk 3]
```diff
@@ -28,6 +28,23 @@ type DeletedDatabaseSpec struct {
...
 	// +optional
 	Recover bool `json:"recover,omitempty"`
+	// Origin Spec to store original database information
+	Origin OriginSpec `json:"origin,omitempty"`
+}
+
+type OriginSpec struct {
+	api.ObjectMeta `json:"metadata,omitempty"`
+	// OriginDatabase Spec to store original database Spec
+	OriginDatabase OriginDatabaseSpec `json:"originDatabase,omitempty"`
```
Comment (TamalSaha) [inline api/deleted_database_types.go]:
```
Spec OriginSpec` `json:"spec,omitempty"`
```

File: api/deleted_database_types.go [hunk 4]
```diff
@@ -28,6 +28,23 @@ type DeletedDatabaseSpec struct {
 	// If true, invoke recover operation
 	// +optional
 	Recover bool `json:"recover,omitempty"`
+	// Origin to store original database information
+	Origin Origin `json:"origin,omitempty"`
```
Comment (aerokite) [inline api/deleted_database_types.go]: Origin Origin `json:"origin,omitempty"`

Is it ok?

Comment (TamalSaha) [inline api/deleted_database_types.go]: Yes

**Pull request merged**